internal/aliasgen/aliasgen.go

```diff
@@ -0,0 +1,423 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aliasgen
+
+import (
+	"fmt"
+	"go/doc"
+	"go/types"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"golang.org/x/tools/go/packages"
+)
+
+const (
+	softLineBreak = 77
+)
+
+// Run generators aliases from the srcDir into the destDir and tidies required
+// files.
+func Run(srcDir, destDir string) error {
+	am, err := createMappings(srcDir)
+	if err != nil {
+		return err
+	}
+	if err := am.WriteAliases(destDir); err != nil {
+		return err
+	}
+	if err := goImports(destDir); err != nil {
+		return err
+	}
+	if err := goModTidy(destDir); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Loads information about a Go package in the specified directory and returns
+// the information required to properly create aliases for the public surface.
+func createMappings(dir string) (*aliasGenerator, error) {
+	log.Printf("creating mappings for: %q", dir)
+	conf := &packages.Config{
+		Mode: packages.NeedName | packages.NeedTypes | packages.NeedDeps | packages.NeedSyntax,
+		Dir:  dir,
+	}
+
+	// Load all package info.
+	pkgs, err := packages.Load(conf)
+	if err != nil {
+		return nil, err
+	}
+	if len(pkgs) != 1 {
+		return nil, fmt.Errorf("found %d packages is %s, expected 1", len(pkgs), dir)
+	}
+	pkg := pkgs[0]
+	am := &aliasGenerator{
+		importPath: pkg.PkgPath,
+		pkg:        strings.TrimSuffix(pkg.Name, "pb"),
+	}
+
+	// Load corresponding documentation.
+	docPkg, err := doc.NewFromFiles(pkg.Fset, pkg.Syntax, pkg.PkgPath)
+	if err != nil {
+		return nil, err
+	}
+	identToDoc := make(map[string]string, len(docPkg.Types))
```

Comment:
Would it also be possible to load docs from `Consts` and `Vars` in order to add similar `Deprecated` labels, such as the deprecation on these consts, for example? https://go.dev/src/os/file.go?h=Deprecated%3A#L87

Reply:
I could add this. The reason I did not is these are not treated special in editor or godoc like methods/types/packages. See the godoc for that example: https://pkg.go.dev/os#pkg-constants. It is just a comment so it would be surfaced by tools.

Reply:
While I was at it just did this for the funcs too since they did not have the warning.